lib/internal/cluster/RoundRobinHandle.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+const assert = require('assert');
+const net = require('net');
+const { sendHelper } = require('internal/cluster/utils');
+const getOwnPropertyNames = Object.getOwnPropertyNames;
+const uv = process.binding('uv');
+
+module.exports = RoundRobinHandle;
+
+function RoundRobinHandle(key, address, port, addressType, fd) {
+  this.key = key;
+  this.all = {};
+  this.free = [];
+  this.handles = [];
+  this.handle = null;
+  this.server = net.createServer(assert.fail);
+
+  if (fd >= 0)
+    this.server.listen({ fd });
+  else if (port >= 0)
+    this.server.listen(port, address);
+  else
+    this.server.listen(address);  // UNIX socket path.
+
+  this.server.once('listening', () => {
+    this.handle = this.server._handle;
+    this.handle.onconnection = (err, handle) => this.distribute(err, handle);
+    this.server._handle = null;
+    this.server = null;
+  });
+}
+
+RoundRobinHandle.prototype.add = function(worker, send) {
+  assert(worker.id in this.all === false);
+  this.all[worker.id] = worker;
+
+  const done = () => {
+    if (this.handle.getsockname) {
+      const out = {};
+      this.handle.getsockname(out);
+      // TODO(bnoordhuis) Check err.
+      send(null, { sockname: out }, null);
+    } else {
+      send(null, null, null);  // UNIX socket.
+    }
+
+    this.handoff(worker);  // In case there are connections pending.
+  };
+
+  if (this.server === null)
+    return done();
+
+  // Still busy binding.
+  this.server.once('listening', done);
+  this.server.once('error', (err) => {
+    // Hack: translate 'EADDRINUSE' error string back to numeric error code.
+    // It works but ideally we'd have some backchannel between the net and
+    // cluster modules for stuff like this.
+    const errno = uv['UV_' + err.errno];
+    send(errno, null);
+  });
+};
+
+RoundRobinHandle.prototype.remove = function(worker) {
+  if (worker.id in this.all === false)
+    return false;
+
+  delete this.all[worker.id];
+  const index = this.free.indexOf(worker);
+
+  if (index !== -1)
+    this.free.splice(index, 1);
+
+  if (getOwnPropertyNames(this.all).length !== 0)
+    return false;
+
+  for (var handle; handle = this.handles.shift(); handle.close());
```

Comment:
I find this kindof for construction hard to read, could you put the empty statement, the `;`, on the next line?